### PR TITLE
show chapters in bookmark dialog

### PR DESCRIPTION
**Pull request by tamland**
This is a stripped down port of @notspiff chapter menu. Removed:
- Thumbs. It's too buggy and extremely slow in its current state. 
- Chapter names. On every file/dvd I could find, chapter name was the time of the chapter (i.e it shows two timestamps in GUI). Changed it to using generic "Chapter i" until a better solution can be found.

I'd prefer this to be done properly in a new window/menu as a simple list, but until somebody does that, this is a working alternative.

### CODE REVIEW

**Comment by tamland** [comment]
@ace20022 Bluray parts could use some testing. I haven't tried it at all with bluray

File: xbmc/video/dialogs/GUIDialogVideoBookmarks.cpp [hunk 1]
```diff
@@ -463,5 +482,3 @@ bool CGUIDialogVideoBookmarks::OnAddEpisodeBookmark()
   }
   return bReturn;
 }
-
```
Comment (ace20022) [inline xbmc/video/dialogs/GUIDialogVideoBookmarks.cpp]: unrelated

File: xbmc/cores/dvdplayer/DVDInputStreams/DVDInputStreamBluray.cpp [hunk 2]
```diff
@@ -558,7 +558,7 @@ void CDVDInputStreamBluray::ProcessEvent() {
     m_player->OnDVDNavResult((void*) &pid, 3);
     break;
 
-#if (BLURAY_VERSION >= BLURAY_VERSION_CODE(0,2,2))
+#if (BLURAY_VERSION > BLURAY_VERSION_CODE(0,2,2))
```
Comment (ace20022) [inline xbmc/cores/dvdplayer/DVDInputStreams/DVDInputStreamBluray.cpp]: unrelated

**Comment by ace20022** [comment]
I'll try to find some time.

File: xbmc/cores/dvdplayer/DVDInputStreams/DVDInputStreamBluray.cpp (hunk 2 above)
Comment (notspiff) [inline xbmc/cores/dvdplayer/DVDInputStreams/DVDInputStreamBluray.cpp]: nope. chapters were exposed in 0.2.3.

Comment (ace20022) [inline xbmc/cores/dvdplayer/DVDInputStreams/DVDInputStreamBluray.cpp]: But that one is ```case BD_EVENT_MENU```

Comment (notspiff) [inline xbmc/cores/dvdplayer/DVDInputStreams/DVDInputStreamBluray.cpp]: right you are. i must have bf'd on a rebase at some point.

Comment (notspiff) [inline xbmc/cores/dvdplayer/DVDInputStreams/DVDInputStreamBluray.cpp]: right. i have squashed an unrelated commit. but the change is correct, BD_EVENT_MENU was exposed in 0.2.3 and is not around in 0.2.2

Comment (ace20022) [inline xbmc/cores/dvdplayer/DVDInputStreams/DVDInputStreamBluray.cpp]: https://github.com/ace20022/libbluray/commit/6d88105783fa3a83963178d31f624717334ca9e0 ;) The one you're referring to is (also) emitted for bd-j menus since 0.30.

**Comment by ace20022** [comment]
> Weird. I have tested mkv and isos (bd and dvd) local and smb. Note i am not questioning the existence of bugs or whatnot just trying to understand the triggers..

Found them, really strange that it works for you ;) Preview: https://github.com/ace20022/xbmc/commits/chapters
Haven't finished reviewing

**Comment by notspiff** [comment]
/confused. you have taken 5 commits from my branch in there, and yet you think it's weird it works in my tree ?
